presto-verifier/src/main/java/com/facebook/presto/verifier/QueryRewriter.java

```diff
@@ -0,0 +1,153 @@
+        this.doublePrecision = doublePrecision;
+    }
+
+    public Query shadowQuery(Query query)
+            throws QueryRewriteException, SQLException
+    {
+        if (statementToQueryType(parser, query.getQuery()) == READ) {
+            return query;
+        }
+        if (!query.getPreQueries().isEmpty()) {
+            throw new QueryRewriteException("Cannot rewrite queries that use pre-queries");
+        }
+        if (!query.getPostQueries().isEmpty()) {
+            throw new QueryRewriteException("Cannot rewrite queries that use post-queries");
+        }
+
+        Statement statement = parser.createStatement(query.getQuery());
+        try (Connection connection = DriverManager.getConnection(gatewayUrl, usernameOverride.orElse(query.getUsername()), passwordOverride.orElse(query.getPassword()))) {
+            if (statement instanceof CreateTableAsSelect) {
+                return rewriteCreateTableAsSelect(connection.getMetaData(), query, (CreateTableAsSelect) statement);
+            }
+        }
+
+        throw new QueryRewriteException("Unsupported query type: " + statement.getClass());
+    }
+
+    private Query rewriteCreateTableAsSelect(DatabaseMetaData metaData, Query query, CreateTableAsSelect statement)
+            throws SQLException
+    {
+        List<String> parts = new ArrayList<>();
+        int originalSize = statement.getName().getOriginalParts().size();
+        int prefixSize = rewritePrefix.getOriginalParts().size();
+        if (originalSize > prefixSize) {
+            parts.addAll(statement.getName().getOriginalParts().subList(0, originalSize - prefixSize));
+        }
+        parts.addAll(rewritePrefix.getOriginalParts());
+        parts.set(parts.size() - 1, rewritePrefix.getSuffix() + UUID.randomUUID().toString().replace("-", ""));
+        QualifiedName temporaryTableName = QualifiedName.of(parts);
+        Statement rewritten = new CreateTableAsSelect(temporaryTableName, statement.getQuery(), statement.isNotExists(), statement.getProperties(), statement.isWithData());
+        String createTableAsSql = formatSql(rewritten);
+        String checksumSql = checksumSql(metaData, catalogOverride.orElse(query.getCatalog()), schemaOverride.orElse(query.getSchema()), statement.getName().getSuffix(), temporaryTableName);
+        String dropTableSql = dropTableSql(temporaryTableName);
+        return new Query(query.getCatalog(), query.getSchema(), ImmutableList.of(createTableAsSql), checksumSql, ImmutableList.of(dropTableSql), query.getUsername(), query.getPassword(), query.getSessionProperties());
+    }
+
+    private String checksumSql(DatabaseMetaData metaData, String catalog, String schema, String existingTable, QualifiedName table)
+            throws SQLException
+    {
+        ResultSet columns = metaData.getColumns(catalog, escapeNamePattern(metaData.getSearchStringEscape(), schema), escapeNamePattern(metaData.getSearchStringEscape(), existingTable), null);
```

Comment:
I wonder if we can structure the create sql into a select statement and then get the result set metadata for the column names.